### PR TITLE
Add Neon implementation of erf()

### PR DESCRIPTION
**Pull request by Rohanjames1997**
The Error function - `erf()` -  is used by the GELU activation function and is a hotspot for many transformer models including BERT.

This NEON implementation of `erf()` improves performance by 50-60%. It is based on Intel's AVX implementation of the same.

### CODE REVIEW

File: aten/src/ATen/cpu/vec/vec256/vec256_float_neon.h [hunk 1]
```diff
@@ -840,6 +835,37 @@ Vectorized<float> inline fmsub(const Vectorized<float>& a, const Vectorized<floa
...
+    // - exp(- x * x)
+    auto pow_2 = (*this) * (*this);
+    auto neg_pow_2 = pow_2 ^ neg_zero_vec;
+    auto tmp4 = neg_pow_2.map(std::exp);
+    auto tmp5 = tmp4 ^ neg_zero_vec;
+    // erf(x) = sign(x) * (1 - r * t * exp(- x * x))
+    auto tmp6 = t * tmp5;
+    auto tmp7 = fmadd(tmp6, r, one_vec);
+    return tmp7 ^ sign_mask;
+}
```
Comment (Rohanjames1997) [inline aten/src/ATen/cpu/vec/vec256/vec256_float_neon.h]: Replica of the [AVX-2 implementation](https://github.com/pytorch/pytorch/blob/fb55f12cb05ddd09bdb385ce9066fa96c20c4320/aten/src/ATen/cpu/vec/vec256/vec256_float.h#L158-L189)

Comment (Rohanjames1997) [reply aten/src/ATen/cpu/vec/vec256/vec256_float_neon.h]: Also note that I've reused the [existing NEON implementations](https://github.com/Rohanjames1997/pytorch/pull/3/files#diff-beea4faf0f4c1a88c5c4db13f8c5e69924c85e3aedc7d5330dd72a397d4639d9R739-R770) of operations like '&', '^', etc.

File: aten/src/ATen/cpu/vec/vec256/vec256_float_neon.h [hunk 2]
```diff
@@ -840,6 +835,37 @@ Vectorized<float> inline fmsub(const Vectorized<float>& a, const Vectorized<floa
...
+    auto t = one_vec / tmp0;
+    // r = p5 * t ^ 4 + p4 * t ^ 3 + p3 * t ^ 2 + p2 * t + p1
+    auto tmp1 = fmadd(p5, t, p4);
+    auto tmp2 = fmadd(tmp1, t, p3);
+    auto tmp3 = fmadd(tmp2, t, p2);
+    auto r = fmadd(tmp3, t, p1);
+    // - exp(- x * x)
+    auto pow_2 = (*this) * (*this);
+    auto neg_pow_2 = pow_2 ^ neg_zero_vec;
+    auto tmp4 = neg_pow_2.map(std::exp);
```
Comment (yd2102) [inline aten/src/ATen/cpu/vec/vec256/vec256_float_neon.h]: Would you like to add a comment saying std::exp can later be swapped with a more optimized version of it?

Comment (Rohanjames1997) [reply aten/src/ATen/cpu/vec/vec256/vec256_float_neon.h]: Yes, good idea. I'll add that comment.

**Pull request closed without merging**